core/src/OpenMP/Kokkos_OpenMP_Parallel_For.hpp

```diff
@@ -202,6 +204,9 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>,
 
  public:
   inline void execute() const {
+    // Serialize kernels on the same execution space instance
+    std::lock_guard<std::mutex> lock(m_instance->m_instance_mutex);
```

Comment:
This is the same approach as for the `Serial` backend. We must ensure kernels don't run concurrently even if submitted by multiple threads and for `ParallelFor<[MD]RangePolicy>` there has not been any synchronization mechanism. In the other cases, we replace the `acquire_loock/release_lock` (initially designed to guard the use of scratch memory) with a `lock_guard` as well for consistency and since we need to control a larger scope.